pkg/metrics/metrics.jsx

```diff
@@ -1667,33 +1676,68 @@ class MetricsHistory extends React.Component {
             );
         }
 
+        const columnVisibilityMenuItems = this.columns.map(itm => {
+            return (
+                <SelectOption
+                    key={itm[0]}
+                    value={itm[1]}
+                    inputId={'column-visibility-option-' + itm[0]} />
+            );
+        });
+        const selections = (
+            this.columns
+                    .filter(itm => this.state.selectedVisibility[itm[0]])
+                    .map(itm => itm[1])
+        );
+        const graphsVisibilitySelector = (
+            <Select
+                toggleAriaLabel={_("Graph visibility options menu")}
+                className="metrics-label"
+                variant={SelectVariant.checkbox}
+                isCheckboxSelectionBadgeHidden
+                isOpen={!!this.state.isOpenColumnVisibility}
+                onSelect={(_, selection) => {
+                    const s = this.columns.find(itm => itm[1] == selection);
+                    this.setState({ selectedVisibility: { ...this.state.selectedVisibility, [s[0]]: !this.state.selectedVisibility[s[0]] } });
+                }}
+                onToggle={() => this.setState({ isOpenColumnVisibility: !this.state.isOpenColumnVisibility })}
+                placeholderText={_("Graph visibility")}
+                selections={selections}>
+                {columnVisibilityMenuItems}
+            </Select>
+        );
+
         return (
             <div className="metrics">
                 <div className="metrics-heading-sticky">
                     <section className="metrics-heading" style={{ "--has-swap": swapTotal ? "var(--column-size)" : "var(--half-column-size)" }}>
-                        <Select
-                            className="select-min metrics-label"
-                            aria-label={_("Jump to")}
-                            onToggle={this.handleToggle}
-                            onSelect={this.handleSelect}
-                            isOpen={this.state.isDatepickerOpened}
-                            selections={this.state.selectedDate}
-                            toggleId="date-picker-select-toggle"
-                        >
-                            {options}
-                        </Select>
+                        <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+                            <Select
+                                className="select-min metrics-label"
+                                aria-label={_("Jump to")}
+                                onToggle={this.handleToggle}
+                                onSelect={this.handleSelect}
+                                isOpen={this.state.isDatepickerOpened}
+                                selections={this.state.selectedDate}
+                                toggleId="date-picker-select-toggle"
+                            >
+                                {options}
+                            </Select>
+                            {graphsVisibilitySelector}
```

Comment:
Feels a bit inconsistent to define the date picker inline here, and the graph visibility selector separately, but no biggie.

Reply:
Moved.